src/operations/client_bulk_write/common.ts

```diff
@@ -0,0 +1,133 @@
+import { type Document } from '../../bson';
+import type { Filter, OptionalId, UpdateFilter, WithoutId } from '../../mongo_types';
+import type { CollationOptions, CommandOperationOptions } from '../../operations/command';
+import type { Hint } from '../../operations/operation';
+
+/** @public */
+export interface ClientBulkWriteOptions extends CommandOperationOptions {
+  /**
+   * If true, when an insert fails, don't execute the remaining writes.
+   * If false, continue with remaining inserts when one fails.
+   * @defaultValue `true` - inserts are ordered by default
+   */
+  ordered?: boolean;
+  /**
+   * Allow driver to bypass schema validation.
+   * @defaultValue `false` - documents will be validated by default
+   **/
+  bypassDocumentValidation?: boolean;
+  /** Map of parameter names and values that can be accessed using $$var (requires MongoDB 5.0). */
+  let?: Document;
+  /**
+   * Whether detailed results for each successful operation should be included in the returned
+   * BulkWriteResult.
+   */
+  verboseResults?: boolean;
+}
+
+/** @public */
+export interface ClientWriteModel {
+  /** The namespace for the write. */
+  namespace: string;
+}
+
+/** @public */
+export interface ClientInsertOneModel<TSchema extends Document = Document>
+  extends ClientWriteModel {
+  name: 'insertOne';
+  /** The document to insert. */
+  document: OptionalId<TSchema>;
+}
+
+/** @public */
+export interface ClientDeleteOneModel<TSchema extends Document = Document>
+  extends ClientWriteModel {
+  name: 'deleteOne';
+  /** The filter to limit the deleted documents. */
+  filter: Filter<TSchema>;
+  /** Specifies a collation. */
+  collation?: CollationOptions;
+  /** The index to use. If specified, then the query system will only consider plans using the hinted index. */
+  hint?: Hint;
+}
+
+/** @public */
+export interface ClientDeleteManyModel<TSchema extends Document = Document>
+  extends ClientWriteModel {
+  name: 'deleteMany';
+  /** The filter to limit the deleted documents. */
```

Comment:
(same suggestion for deleteOne)
```suggestion
  /** 
   * The filter used to determine if a document should be deleted. For a deleteOne operation, the first match is removed.
  */
```

Reply:
(also maybe a similar change for the updateOne updateMany models)

Reply:
Updated all comments.